### DISCUSSION TITLE
Specify type as relative pointer to structure

### DISCUSSION THREAD

**Discussion by MrQubo** [Vector35/binaryninja-api#4750] [Q&A] [answered]
The code I'm reversing takes address of the structure + 0x10 and, probably because of some optimizations, uses this relative pointer to access structure fields. Is it possible to have this displayed correctly in decompilation?

Iirc ghidra allows creating relative pointer typedefs for that. You can create a type for a pointer with offset.

**Comment by alexrp** [comment, marked as answer]
You probably want to use either `__offset(T, N)` on the parameter/variable or `__ptr_offset(N)` on the struct definition.

**Reply by fuzyll** [reply]
For additional context, we released this in 3.2 and [this](https://binary.ninja/2022/10/28/3.2-released.html#offset-pointers) is the relevant part of the release notes, in case anyone wants a much lengthier answer.

**Comment by MrQubo** [comment]
Works perfectly. :D
Thank you!
